server/apps/packages/package_service.py

```diff
@@ -266,3 +278,12 @@ def remove_refs_in_package(self, package, ref_id_to_remove, processed_packages=N
     def _get_associations(self, doc):
         return [assoc for group in doc.get('groups', [])
                 for assoc in group.get(ASSOCIATIONS, [])]
+
+    def remove_spiked_refs_from_package(self, doc_id):
+        packages = self.get_packages(doc_id)
+        if packages.count() > 0:
```

Comment:
Isn't this redundant? If the package count is zero, the `package in packages` loop won't run anyway.

(well, OK, it does save one list instantiation...)